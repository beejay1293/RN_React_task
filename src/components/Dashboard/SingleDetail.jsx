import React from 'react';
import propTypes from 'prop-types';
import { Col, Card, CardTitle, CardText, Table } from 'reactstrap';

const SingleDetail = ({
  name, updated, pchange1, pchange2, pchange3, price, created, sym, rank, supply
}) => (
  <Col sm="4">
  <Card body className="text-center DetailsCard">
    <CardTitle>{name} ({sym})</CardTitle>
    <p>Percentage Changes</p>
    <Table hover>
        <thead>
          <tr>
            <th>1 Hour</th>
            <th>27 Hours</th>
            <th>7 days</th>

          </tr>
        </thead>
        <tbody>
          <tr>
            <td>{pchange1}</td>
            <td>{pchange2}</td>
            <td>{pchange3}</td>
          </tr>

        </tbody>
      </Table>
      <CardTitle>Price: {price}</CardTitle>
      <CardTitle>Total Supply: {supply}</CardTitle>
      <CardTitle>cmc Rank: {rank}</CardTitle>

    <CardText><b>Added on</b> {created}.</CardText>
    <CardText><b>Updated on</b> {updated}.</CardText>

  </Card>
</Col>
);

SingleDetail.propTypes = {
  name: propTypes.string.isRequired,
  price: propTypes.number.isRequired,
  rank: propTypes.number.isRequired,
  supply: propTypes.number.isRequired,
  pchange1: propTypes.number.isRequired,
  pchange2: propTypes.number.isRequired,
  pchange3: propTypes.number.isRequired,  
};
export default SingleDetail;
