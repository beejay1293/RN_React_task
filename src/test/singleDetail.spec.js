import React from 'react';
import { shallow } from 'enzyme';
import  SingleDetail from '../components/Dashboard/SingleDetail';

describe('test Single Detail component', () => {
  it('should render component', () => {
      const prop = {
        name: "RN", 
        supply: 300, 
        price: 20, 
        rank:3,
        pchange1: 23, 
        pchange2: 455, 
        pchange3: 95604
      }
    const SDetail = shallow(<SingleDetail { ...prop} />);
    expect(SDetail).toMatchSnapshot();
  });
});
