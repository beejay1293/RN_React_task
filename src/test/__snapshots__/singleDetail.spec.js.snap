// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Single Detail component should render component 1`] = `
<Col
  sm="4"
  tag="div"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ]
  }
>
  <Card
    body={true}
    className="text-center DetailsCard"
    tag="div"
  >
    <div
      className="SingleList"
    >
      <div>
        <div />
      </div>
      <div />
    </div>
    <CardTitle
      tag="div"
    >
      RN
       (
      )
    </CardTitle>
    <p>
      Percentage Changes
    </p>
    <Table
      hover={true}
      responsiveTag="div"
      tag="table"
    >
      <thead>
        <tr>
          <th>
            1 Hour
          </th>
          <th>
            27 Hours
          </th>
          <th>
            7 days
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            23
          </td>
          <td>
            455
          </td>
          <td>
            95604
          </td>
        </tr>
      </tbody>
    </Table>
    <CardTitle
      tag="div"
    >
      Price: 
      20
    </CardTitle>
    <CardTitle
      tag="div"
    >
      Total Supply: 
      300
    </CardTitle>
    <CardTitle
      tag="div"
    >
      cmc Rank: 
      3
    </CardTitle>
    <CardText
      tag="p"
    >
      <b>
        Added on
      </b>
       
      .
    </CardText>
    <CardText
      tag="p"
    >
      <b>
        Updated on
      </b>
       
      .
    </CardText>
  </Card>
</Col>
`;
